### PR TITLE
Fix NSJSONSerialization JSONObjectWithData tests to only verify error code  and not compare description string

### PR DESCRIPTION
**Pull request by mnithish**

### CODE REVIEW

**Comment by mnithish** [comment]
@DHowett-MSFT is added to the review. #Closed

**Comment by mnithish** [comment]
@ms-jihua is added to the review. #Closed

**Comment by DHowett-MSFT** [comment]
:shipit:

File: tests/unittests/Foundation/NSJSONSerializationTests.m [hunk 1]
```diff
@@ -106,12 +97,11 @@ void VerifyDataWithJSONObjectThrows(id testObject) {
     VerifyJSONObjectWithDataSucceeds(testString6, 1, @{ @"foo" : @"1", @"bar" : @"2" });
 
     // error cases
-    NSString* fragmentDescription = @"JSON text did not start with array or object and option to allow fragments not set.";
```
Comment (mnithish) [inline tests/unittests/Foundation/NSJSONSerializationTests.m]: >@"JSON text did not start with array or object and option to allow fragments not set." [](start = 36, length = 86)

Reference platform does not document these strings and it seems like an over kill to compare error description in UT. Error codes should be sufficient

Note: An error code can return different descriptive descriptions, so error code to description is not an 1:1 mapping always

**Comment by rajsesh-msft** [comment]
:shipit:

**Pull request merged**